### ISSUE TITLE
Matching expectation for mocks output for function with Variadic inputs

### ISSUE THREAD

**Issue by vish91** [vektra/mockery#215]
I have a function that takes in a slice as an argument
```
func (b *batchFirer) Fire(ctx context.Context, id string, batch ...Items) {

```
for which I see a difference between how `latest` generates the mock and how an older version generated mock for it. 
new version : 
```
func (_m *BatchFirer) Fire(_a0 context.Context, _a1 string, _a2 ...dao.Items) {
	_va := make([]interface{}, len(_a2))
	for _i := range _a2 {
		_va[_i] = _a2[_i]
	}
	var _ca []interface{}
	_ca = append(_ca, _a0, _a1)
	_ca = append(_ca, _va...)
	_m.Called(_ca...)
}
```
older version: 
```
func (_m *BatchFirer) Fire(_a0 context.Context, _a1 string, _a2 ...dao.Items) {
	_m.Called(_a0,_a1,_a2)
}
```

^ this cause issues because now where this mock is being called appends slice as an argument, but the expected assertion is expecting individual elements as arguments
```
listOfItems := []Items{"a","b",}
mockItems.On("Fire", mock.Anything, resp.ID, listOfItems)
```
Example error:
```
mock: Unexpected Method Call
-----------------------------

Fire(*context.valueCtx,string,dao.Items,dao.Items)
		0: &context.valueCtx{Context:(*context.valueCtx)(0xc00013fce0), key:"account_id", val:"account_1"}
		1: "abc"
		2: dao.Items{"a"}
		3: dao.Items{"b"}

The closest call I have is: 

Fire(string,string,[]dao.Items)
		0: "mock.Anything"
		1: "abc"
		2: []dao.Items{"a","b"}

Provided 3 arguments, mocked for 4 arguments
......	panic: 
```

Cannot change 
`mockItems.On("Fire", mock.Anything, resp.ID, listOfItems)` to 
`mockItems.On("Fire", mock.Anything, resp.ID, listOfItems...)` because `fire` function expects []Items and not x Items

Is there something that can be done to fix this situation of a slice being mocked to a variadic input, and mock function expecting 0..n string arguments being passed on mock call instead of expecting a []string as argument.

**Comment by awishformore** [comment]
I just ran into this same issue. You can't expand the slice with `...` if it's not the first variadic parameter of a function. Since `On()` is variadic itself, you have to explicitly provide each value, which doesn't work if you have a generated slice.

A possible (annoying) workaround for a function `Func(first int, second ...int)` is:
```go
params := []interface{first}
for _, v := range second {
   params = append(params, v)
}
m.On("Func", params...).Return(nil)
```

**Comment by LandonTClipp** [comment]
This is indeed an issue. This issue was listed before we started following semantic versioning and is quite old, so in the interest of pruning possible stale issues, I'm going to close this. If this is still presenting a problem for people please open a new issue and reference this one.
